Feature/peer

paulfariello opened the pull request.

paulfariello committed:
Add support for peering between stormbot

If there is multiple stormbot (with similar names) in a muc,
they will try to exchange informations about their plugins.

Then a master (the one with shortest name) will forward every requests
to others stormbot.

paulfariello committed:
Send iq response on command reception

paulfariello committed:
Use feature discovery

Store list of plugin in a disco node.
Identify peer with a feature in disco.

Switch to slixmpp because had issue with sleekxmpp and the later is not
maintained anymore.

paulfariello committed:
Allow each plugins to decide weither command must be forwarded or not

Plugin have read access on peer list and have to loop over each
concerned peer and for each of them call self._bot.peer_send_command.

Beware of using shlex.quote if forging raw command manually.

paulfariello committed:
Use twine to check restructuredtext

paulfariello committed:
Bump dev version

paulfariello committed:
Add travis support

paulfariello committed:
Use loose version checking

paulfariello committed:
Bump dev version

paulfariello committed:
Add support for dev version in Makefile

paulfariello committed:
Fix docker README

paulfariello committed:
Add tips about audio device for docker run

paulfariello committed:
Use nosetest for tests in travis-ci

paulfariello committed:
Only publish on tags

paulfariello committed:
Fix typo in travis provider

paulfariello committed:
Set travis script to build, twine check and nosetests

paulfariello committed:
Add version plugin

paulfariello committed:
Bump dev version

paulfariello committed:
Fix command line running

paulfariello committed:
Add mocking decorator for unit tests

paulfariello committed:
Bump dev version

SabatierBoris commented on .travis.yml:
```diff
@@ -0,0 +1,19 @@
+python:
+  - "3.6"
+install:
+  - pip install nose twine
```
Why nose ?
Why not use develop_dependencie in the setup.py ?
If you still want to use nose, you can say it in the setup.py so you can run `./setup.py test`

paulfariello replied:
I could remove nose and just use unittest.

SabatierBoris commented on stormbot/storage.py:
```diff
@@ -68,5 +71,17 @@ def proxy(self, value):
+        self._file.seek(0)
+        json.dump(self._cache, self._file, cls=ProxyEncoder)
+
+def mock(values):
```
Why this code isn't in the test directory ?

SabatierBoris commented on stormbot/bot.py:
```diff
@@ -162,19 +275,156 @@ def subscribe(self, nick, plugin):
     def write(sef, *args, **kwargs):
         print(*args, **kwargs)
 
+def mock(argv):
```
Why this code isn't in the test directory ?

paulfariello committed:
Add storage tests

paulfariello committed:
Add docker build on tag starting with docker-

paulfariello committed:
Try to fix travis docker deployment

paulfariello committed:
Add support for command response

paulfariello committed:
Print stack trace before sending command error

paulfariello committed:
Fix storage initialization

Force dump of initial storage because empty file is not a valid json.
Force flush when dumping storage because json might not contains
newlines.

paulfariello committed:
Bump dev version

paulfariello committed:
Bump dev version

paulfariello committed:
Fix commandline helper for plugin with no run function

paulfariello committed:
Bump dev version

paulfariello committed:
Rework tests

Get rid of nosetest
Move mock to stormbot.mock

paulfariello committed:
Truncate storage file when dumping storage

paulfariello committed:
Update version and help to async command

paulfariello committed:
Allow to test with patched StormBot instead of patched FakeBot

See stormbot-quote for an example of unit tests

paulfariello committed:
Add temporary dev version in dockerfile
